app/controllers/forgot_passwords_controller.rb

```diff
@@ -0,0 +1,42 @@
+class ForgotPasswordsController < ApplicationController  
+  skip_before_action :require_user
+
+  def create
+    user = User.find_by(email: params[:email])
+    if user
+      user.generate_token!
+      token = user.reload.token
```

Comment:
Do you have to reload? I would think that `update` would affect the in-memory object and you would already have the token attached to the object.